dotnet-desktop-guide/winforms/controls/snippets/how-to-make-thread-safe-calls/cs/InvokeAsyncExamples.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Net.Http;
+using System.Threading; // Added for CancellationToken
+using System.Threading.Tasks;
+using System.Windows.Forms;
+
+namespace project;
+
+public partial class InvokeAsyncForm : Form
+{
+    private Button button1;
+    private Button button2;
+    private TextBox textBox1;
+
+    public InvokeAsyncForm()
+    {
+        button1 = new Button { Text = "Invoke Async Basic", Location = new System.Drawing.Point(10, 10), Width = 200 };
+        button2 = new Button { Text = "Invoke Async Advanced", Location = new System.Drawing.Point(10, 50), Width = 200 };
+        textBox1 = new TextBox { Location = new System.Drawing.Point(10, 90), Width = 200, Multiline = true, Height =500 };
+        button1.Click += button1_Click;
+        button2.Click += button2_Click;
+        Controls.Add(button1);
+        Controls.Add(button2);
+        Controls.Add(textBox1);
+    }
+
+    // <snippet_InvokeAsyncBasic>
+    private async void button1_Click(object sender, EventArgs e)
+    {
+        button1.Enabled = false;
+        
+        try
+        {
+            // Perform background work
+            await Task.Run(async () =>
+            {
+                for (int i = 0; i <= 100; i += 10)
+                {
+                    // Simulate work
+                    await Task.Delay(100);
```

Comment:
Maybe we're using something like Thread.Spin to actually simulate work?
Task.Delay doesn't do any "work".